broker/broker.go

```diff
@@ -53,16 +46,36 @@ var composeStatus2State = map[string]brokerapi.LastOperationState{
 	"waiting":  brokerapi.InProgress,
 }
 
-func New(compose compose.Client, config *config.Config, catalog *catalog.ComposeCatalog, logger lager.Logger) (*Broker, error) {
-	if compose == nil {
-		return nil, fmt.Errorf("broker: composer should not be nil")
+type Broker struct {
+	Compose        compose.Client
+	Config         *config.Config
+	ComposeCatalog *catalog.ComposeCatalog
+	Logger         lager.Logger
+	AccountID      string
+	ClusterID      string
```

Comment:
I was thinking that Config should deal with all of the config that we explicitly set. It feels odd to have Config have to deal with calling out to Compose etc. I therefore decided to move this into the Broker constructor. As a result, I think these values belong in the broker data structures, and not in config.